src/Common/src/System/LocalAppContext.cs

```diff
@@ -8,6 +8,15 @@ namespace System
 {
     internal partial class LocalAppContext
     {
+        static LocalAppContext()
+        {
+            bool isEnabled;
+            if (AppContext.TryGetSwitch(@"TestSwitch.LocalAppContext.DisableCaching", out isEnabled))
+            {
+                DisableCaching = isEnabled;
+            }
+        }
```

Comment:
It's best to avoid explicit static cctors when possible.  This would be better as:
```C#
private static bool GetDisableCachingSwitch()
{
    bool isEnabled;
    return AppContext.TryGetSwitch(@"TestSwitch.LocalAppContext.DisableCaching", out isEnabled) && isEnabled;
}

private static bool DisableCaching { get; } = GetDisableCachingSwitch();
```
or with C# 7:
```
private static readonly bool s_disableCaching = AppContext.TryGetSwitch(@"TestSwitch.LocalAppContext.DisableCaching", out bool isEnabled) && isEnabled;
```